gamemodes/terrortown/gamemode/shared/sh_shopeditor.lua

```diff
@@ -75,39 +152,40 @@ function ShopEditor.WriteItemData(messageName, name, item, plys)
 
 	if not name or not item then return end
 
-	net.Start(messageName)
-	net.WriteString(name)
-
 	for key, data in pairs(ShopEditor.savingKeys) do
+		net.Start(messageName)
+		net.WriteString(name)
+		net.WriteString(key)
```

Comment:
That's a ton of nw-messages, what are the reasons changing this?

Reply:
Mainly because of the single variable change in the new UI compared to the bunch of Saves of the old one. I rewrote the send and save method so it would fit.

Reply:
But it was difficult to keep the overview and contain the old structure, so maybe this approach isnt ideal. Actually instead of the old hardcoded number of saves, i could make an individual message size. That would work too.

Reply:
okay, I have to agree with Alf. Why do you send a new message for every key and not one message for every item?

Reply:
Because of ReadItemData. Its used for several use-case and therefore is unified into one function. Its optimized for the new UI, but as I said, I came up with the idea to make that message size variable. You just write and additional parameter that contains the number of keys to save. Then you can have the old system still with all keys and the new system with only one key per save.

In general it's just a bit annoying to keep an old system intact, so I just changed it to thos inefficient way. But maybe I'll need the upper function anyway and then an optimization should be one message per item instead.
Gotta look more closely at the code to be sure :)

Reply:
Ah I understand. So in the new system you mostly send only one key so it is already the most performant way. Only in the old UI it has an overhead. Did I get this right?

Reply:
Yes, in the old UI it's definitely bad and it is optimized for the new UI one way (Client->Server), but what I forgot was, that you send data back to the clients and maybe there I didnt change the system to one key per message. So need to look at that anyway. :)

So definite answer is, I need to check this part too, as it might be still needed for the new UI and therefore produces too many

Reply:
It is ok for me then since the focus should be on the new UI anyway

Reply:
https://github.com/TTT-2/TTT2/pull/816/files#diff-eb89179dfc4da23810ef7f998dd5d73c8c3b380007153cbcf4bca2301f451692R155

That means you are looping through every key in ShopEditor every time `ShopEditor.WriteItemData` is called:
https://github.com/TTT-2/TTT2/pull/816/files#diff-eb89179dfc4da23810ef7f998dd5d73c8c3b380007153cbcf4bca2301f451692R6-R67

If that's right and intended, then feel free merging it :)